Zenodo tests fail

The zenodo support added in #775 doesn't pass tests on my computer.

* papis version (`$ papis --version` or commit number): ee68780e

Dependency versions on my system (Arch Linux)
```
python 3.11.8-1
python-arxiv 2.1.0-1
python-beautifulsoup4 4.12.3-1
python-bibtexparser 1.4.1-2
python-chardet 5.2.0-1
python-click 8.1.7-1
python-colorama 0.4.6-2
python-doi 0.2.0-1
python-dominate 2.9.1-1
python-filetype 1.2.0-2
python-habanero 1.2.3-2
python-isbnlib 3.10.14-2
python-lxml 4.9.2-3
python-prompt_toolkit 3.0.43-1
python-pyaml 23.9.0-2
python-pygments 2.17.2-1
python-pyparsing 3.1.2-1
python-requests 2.31.0-1
python-slugify 8.0.4-1
python-stevedore 5.2.0-1
python-tqdm 4.66.2-1
python-typing_extensions 4.10.0-1
```

# how to reproduce the issue
From a clean checkout, run `python -m pytest papis tests`. Log attached.
[errors.log](https://github.com/papis/papis/files/14573202/errors.log)

Judging by the output, it seems either the plugin or the golden output has stripped the HTML of the exported citation.

> Ah sorry, marked success too soon -- the fix breaks the `mypy` tests
> 
> ```
> papis/zenodo.py:43: error: Unused "type: ignore" comment  [unused-ignore]
>             from markdownify import markdownify  # type: ignore[import-untyped]
>             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> papis/zenodo.py:43:1: error: Cannot find implementation or library stub for module named "markdownify" 
> [import-not-found]
>             from markdownify import markdownify  # type: ignore[import-untyped]
>     ^
> papis/zenodo.py:43:1: note: Error code "import-not-found" not covered by "type: ignore" comment
> papis/zenodo.py:43:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
> Found 2 errors in 1 file (checked 118 source files)
> ```

Hm, what version of `mypy` are you using? I tried with mypy 1.9.0 and I'm not getting any errors.

So testing with `1.10.0+dev.ea49e1fa488810997d192a36d85357dadb4a7f14` silences that error (and implies we probably should bump the minimal `mypy` version in deps), but raises another one:
```
papis/format.py:90:41: error: Argument 2 of "convert_field" is incompatible with supertype "Formatter";
supertype defines the argument type as "str | None"  [override]
        def convert_field(self, value: Any, conversion: str) -> Any:
                                            ^~~~~~~~~~~~~~~
papis/format.py:90:41: note: This violates the Liskov substitution principle
papis/format.py:90:41: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
Found 1 error in 1 file (checked 118 source files)
```

> ```
> papis/format.py:90:41: error: Argument 2 of "convert_field" is incompatible with supertype "Formatter";
> supertype defines the argument type as "str | None"  [override]
>         def convert_field(self, value: Any, conversion: str) -> Any:
>                                             ^~~~~~~~~~~~~~~
> papis/format.py:90:41: note: This violates the Liskov substitution principle
> papis/format.py:90:41: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
> Found 1 error in 1 file (checked 118 source files)
> ```

Hm, that seems to fail on 1.9.0, so maybe some stubs will get updated too with 1.10.0. In my experience, it doesn't seem possible to have things pass across version of mypy, so I'm generally happy if the CI passes :(

--- a/papis/zenodo.py
+++ b/papis/zenodo.py
@@ -40,11 +40,9 @@ def get_text_from_html(html: str) -> str:
     :return: Either the raw HTML as text, or the markdown-annotated plain text.
     """
     try:
-        from markdownify import markdownify  # type: ignore[import-untyped]
+        from markdownify import markdownify
 
-        result = markdownify(html)  # type: str
-        return result
-
+        return str(markdownify(html))
     except ImportError:
         logger.info(
             "Saving text as raw HTML.  Install `markdownify` to convert it to markdown."

--- a/setup.cfg
+++ b/setup.cfg
@@ -57,5 +57,8 @@ ignore_missing_imports = True
 [mypy-whoosh.*]
 ignore_missing_imports = True
 
+[mypy-markdownify.*]
+ignore_missing_imports = True
+
 [mypy-sphinx_click.*]
 ignore_missing_imports = True

